### DISCUSSION TITLE
RQL code to create an multi map index simulating a join query

### DISCUSSION THREAD

**Discussion by dbblackdiamond** [ravendb/ravendb#18269] [Q&A] [answered]
Hi,

even though, I have read all the blogs and post on how to simulate join queries on ravendb, I am stumped. 
I have the following entities:
```
race {
eventid,
status
}
entry {
userid,
raceid
}
```

The field `raceid` in the entity `entry` is a reference to the `eventid` in the `race` entity. I am trying to create an index that allows me to all the races a particular user is registered in and that have a specific `status` .  I have seen some code that come close to achieving that, but all of those examples are in LINQ and I don't know C#, so how do I do this in RQL?
I have tried creating various types of map-reduce indexes to no avail as I always seem to be getting an error. My last attempt was trying to replicate this:
```
public class Employees_ByDepartment : AbstractMultiMapIndexCreationTask<Employees_ByDepartment.Entry>
{
    public class Entry
    {
        public string[] EmployeeId { get; set; }

        public string StoreId { get; set; }

        public string DepartmentId { get; set; }
    }

    public Employees_ByDepartment()
    {
        AddMap<Department>(departments =>
            from department in departments
            from storeId in department.StoreIds
            select new Entry
            {
                EmployeeId = new string[] {},
                StoreId = storeId,
                DepartmentId = department.Id
            });

        AddMap<Employee>(employees =>
            from employee in employees
            select new Entry
            {
                EmployeeId = new string[] { employee.Id },
                StoreId = employee.StoreId,
                DepartmentId = ""
            });

        Reduce = results =>
            from result in results
            group result by new
            {
                result.StoreId
            }
            into g
            select new Entry
            {
                StoreId = g.Key.StoreId,
                EmployeeId = g.SelectMany(x => x.EmployeeId).ToArray(),
                DepartmentId = g.First(x => x.DepartmentId != "").DepartmentId
            };
    }
}
``` 
in ravendb studio by defining some fields and then the 2 maps and 1 reduce, but I got a syntax error, so how do I do this in RQL? Ideally, the index would return the entire `race` entity.

Thanks a lot,
Bertrand.

**Comment by ppekrol** [comment, marked as answer]
Hi,

I'm wondering if you cannot solve this via a LoadDocument and a single map index, sth like:

![image](https://github.com/ravendb/ravendb/assets/446555/12adb600-71ce-4967-a73b-8c486196da30)

(please mind collection names and that I've set raceid as a stored field, so projection will extract the value from the index directly, not from a document)

and then query it via:
```
from index 'IndexName'
where userid = 'users/1' and status = 'completed'
select raceid
```

Would that work for you?

**Reply by dbblackdiamond** [reply]
It totally worked. I have been struggling with the syntax of the `LoadDocument` instruction for a while, so thank you for showing me this.
